service-base/src/test/java/org/eclipse/hono/service/credentials/AbstractCompleteCredentialsServiceTest.java

```diff
@@ -51,12 +50,10 @@ public void testAddCredentialsRefusesDuplicateRegistration(final TestContext ctx
                 .put(CredentialsConstants.FIELD_AUTH_ID, "myId")
                 .put(CredentialsConstants.FIELD_TYPE, "myType")
                 .put(CredentialsConstants.FIELD_SECRETS, new JsonArray());
-        final Async add = ctx.async();
-        getCompleteCredentialsService().add("tenant", payload2, ctx.asyncAssertSuccess(s -> {
-            assertThat(s.getStatus(), is(HttpURLConnection.HTTP_CONFLICT));
-            add.complete();
-        }));
-        add.await();
+        getCompleteCredentialsService().add("tenant", payload2, ctx.succeeding(s -> ctx.verify(() -> {
+            assertEquals(HttpURLConnection.HTTP_CONFLICT, s.getStatus());
```

Comment:
what's wrong with using `assertThat`? FMPOV it is easier to read and produces better error messages...

Reply:
`assertThat` is removed from junit5. IMHO, for comparing HTTP status codes `assertEquals` is more intuitive.

Reply:
just wanted to make sure that we do not strictly replace all occurrences of `assertThat` with `assertEquals` because we think that we need to.

Reply:
We absolutely don't have to. As long as `hamcrest-core` stays in the dependencies that will work fine. I simply wanted to try it because using the built-in assertions would remove a dependency. And also for the sake of trying something new :)